### PR TITLE
[Mellanox] Support one ingress pool mode

### PR DESCRIPTION
**Pull request by stephenxs**
Support one ingress pool mode.

Signed-off-by: Stephen Sun <stephens@mellanox.com>

**- Why I did it**
Support one ingress pool mode

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by liat-grozovik** [comment]
retest vsimage please

**Comment by neethajohn** [comment]
According to the SAI spec, the buffer pool size includes the reserved buffer for all pg/queues. So that would mean we would have the same pool size specified for T1 and T0 templates and internally subtract the reserved space to get the actual shared pool size to allocate.
This is followed by other ASICs which do not include a headroom pool
https://github.com/opencomputeproject/SAI/blob/master/doc/QOS/SAI-Proposal-buffers-Ver4.docx

File: build_image.sh [hunk 1]
```diff
@@ -150,7 +150,15 @@ elif [ "$IMAGE_TYPE" = "aboot" ]; then
...
-        [ -f "$CA_CERT" ] && cp "$CA_CERT" "$TARGET_CA_CERT"
+
+        # If the ca certificate does not exist, the test certificate will be used to sign the image
+        if [ ! -f "$CA_CERT" ]; then
+            TEST_CERT_PATH=files/image_config/secureboot/test-certs
+            CA_CERT="${TEST_CERT_PATH}/ca.cert"
+            SIGNING_KEY="${TEST_CERT_PATH}/signing.key"
+            SIGNING_CERT="${TEST_CERT_PATH}/signing.cert"
+        fi
+        cp "$CA_CERT" "$TARGET_CA_CERT"
```
Comment (neethajohn) [inline build_image.sh:161]: why are these changes in here?

File: files/image_config/secureboot/test-certs/signing.cert [hunk 2]
```diff
@@ -0,0 +1,30 @@
+-----BEGIN CERTIFICATE-----
```
Comment (neethajohn) [inline files/image_config/secureboot/test-certs/signing.cert:1]: What is this change?

File: files/image_config/secureboot/test-certs/ca.cert [hunk 3]
```diff
@@ -0,0 +1,32 @@
+-----BEGIN CERTIFICATE-----
```
Comment (neethajohn) [inline files/image_config/secureboot/test-certs/ca.cert:1]: What is this change?

File: files/image_config/secureboot/test-certs/signing.key [hunk 4]
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (neethajohn) [inline files/image_config/secureboot/test-certs/signing.key:1]: What is this change?

File: build_image.sh (hunk 1 above)
Comment (stephenxs) [reply build_image.sh:161]: my bad. it was introduced by mistake once I refuse the PR.
will remove it.

**Comment by stephenxs** [comment]
> ingress_lossy_profile
Hi,
`ingress_lossy_profile` is referenced in `BUFFER_PG|<port>|0`. Please check [sonic-buildimage/build_templates/buffers_config.j2](https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/buffers_config.j2#L136)

**Comment by baiwei0427** [comment]
Do you have buffer templates for 2700 C32 sku?

**Comment by stephenxs** [comment]
> Why different SKUs get different Xoff sizes under the same link speed and cable length?

Do you mean

**Comment by stephenxs** [comment]
> Do you have buffer templates for 2700 C32 sku?

**Comment by stephenxs** [comment]
> Do you have buffer templates for 2700 C32 sku?

Yes. All the buffer templates of SKUs based 2700 share the same templates.
We can provide the specific template for C32 if you provide the cable lengths of ports for t0/t1.
Are they:
- t0: 28 * server ports (5m) + 4 leaf ports (40m)
- t1: 24 * TOR ports (40m) + 8 spine ports (300m)
?

**Comment by baiwei0427** [comment]
> > Why different SKUs get different Xoff sizes under the same link speed and cable length?
> 
> If they are based on different ASICs, they can have different xoff sizes because of different cell sizes.
> If you're not meaning this, can you elaborate it a bit more?

Got it! What are the cell sizes of 3800 and 4600?

**Comment by baiwei0427** [comment]
> > Do you have buffer templates for 2700 C32 sku?
> 
> Yes. All the buffer templates of SKUs based 2700 share the same templates.
> We can provide the specific template for C32 if you provide the cable lengths of ports for t0/t1.
> Are they:
> 
> * t0: 28 * server ports (5m) + 4 leaf ports (40m)
> * t1: 24 * TOR ports (40m) + 8 spine ports (300m)
> 
> ?
Let's assume the worst case.
t0: 16 * server ports (100g, 5M) + 16 leaf ports (100g, 40m)
t1:  16 * TOR ports (100g, 40m) + 16 spine ports (100g, 300m)

**Comment by stephenxs** [comment]
> > Let's assume the worst case.
> > t0: 16 * server ports (100g, 5M) + 16 leaf ports (100g, 40m)
> > t1:  16 * TOR ports (100g, 40m) + 16 spine ports (100g, 300m)
Hi @liat-grozovik @lguohan , If we all agress on this setup I will calculate buffer pool size for `Mellanox-SN2700` based on this.

**Comment by stephenxs** [comment]
> D48C8 and C32 should use different T1 templates

Done. Please review.

**Comment by liat-grozovik** [comment]
@baiwei0427 are we good to go with these values?

**Comment by baiwei0427** [comment]
> @baiwei0427 are we good to go with these values?

I will do calculations tomorrow and get back to you as soon as possible,

**Comment by stephenxs** [comment]
@baiwei0427 could you also provide the cable length information for calculating pool size for 3800 as well?
For 3800, we have the following SKUs:
- Mellanox-SN3800-C64
- Mellanox-SN3800-D112C8
- Mellanox-SN3800-D24C52
- Mellanox-SN3800-D28C50
thank you!

**Comment by baiwei0427** [comment]
> @baiwei0427 could you also provide the cable length information for calculating pool size for 3800 as well?
> For 3800, we have the following SKUs:
> 
> * Mellanox-SN3800-C64
> * Mellanox-SN3800-D112C8
> * Mellanox-SN3800-D24C52
> * Mellanox-SN3800-D28C50
> 
> thank you!

Server to T0: 5M
T0 to T1: 40M
T1 to T2: 300M

**Comment by stephenxs** [comment]
@baiwei0427 
> Server to T0: 5M
> T0 to T1: 40M
> T1 to T2: 300M
How many ports are connected to servers, TORs, leaf/spine routers respectively in each of the SKU?
Thanks

**Comment by baiwei0427** [comment]
@liat-grozovik  @stephenxs  I did calculations for D48C8 T0, SN2700 T0, and SN2700 T1 and got similar results.

**Comment by liat-grozovik** [comment]
Note: Leaving aside new requirements for SN3800. this PR is for updating SN2700 SKUs.

**Comment by abdosi** [comment]
cherry-pick has conflict. Create PR for 201911

@liat-grozovik @rlhui

**Comment by stephenxs** [comment]
> cherry-pick has conflict. Create PR for 201911
> 
> @liat-grozovik @rlhui

@abdosi Done [PR #5194](https://github.com/Azure/sonic-buildimage/pull/5194)

**Pull request merged**